### DISCUSSION TITLE
Error in filtering

### DISCUSSION THREAD

**Discussion by dharik-arsath** [vaexio/vaex#1420] [Q&A]
I have some empty strings and so I used df = df[~(df.x == ' ')]. Now if I check that column things are gone and the data is clean but the column in the string datatype so if I changed that info float it and things are alright here. but when I do some filtering on that column it is throwing me the error that column has empty string. Though I have cleaned and reassigned with df.

I can able to plot histogram, I can able to see even minmax, but when I filter that column it is throwing me the error.What is the problem here? I have tried to materialize as well that also throws me the same error. Please look into the issue. Thank you

When I check for df.x it is giving me the expression with float 32 which is as expected but why can't I filter it like df[df.x > 50000]

**Comment by JovanVeljanoski** [comment]
Can you post a reproducible example?

**Reply by dharik-arsath** [reply]
data[data.TotalCharges == ' ']
This line shows me that there are 10 lines of empty strings and I removed by data = data[~(data.TotalCharges == ' ')]
now if I check data[data.TotalCharges == ' '] it shows nothing because I have reassigned. and then I do type conversion
data['TotalCharges'] = data.TotalCharges.astype('float32')
no error here as well,then If i ask for data.TotalCharges -> it gives expression with float 32. data.TotalCharges.minmax() works as well.

but if I do data[data.TotalCharges > 5000] it shows me long list of error with last line 'ArrowInvalid: Failed to parse string: ' ' as a scalar of type float'

it says failed to parse string but I have already done data = data[~(data.TotalCharges == '  ')]. and changed that to float32

**Reply by dharik-arsath** [reply]
#!pip install vaex -qq
data = vaex.read_csv('../input/telco-customer-churn/WA_Fn-UseC_-Telco-Customer-Churn.csv')

data[data.TotalCharges == ' ']  -> shows me records with empty strings
data = data[~(data.TotalCharges == ' ')] -> reassigns data with no empty strings
data[data.TotalCharges == ' '] -> no more empty strings here
data['TotalCharges'] = data.TotalCharges.astype('float32') -> since there was empty strings by default datatype  was string so I changed to float it works,

data.TotalCharges.minmax() -> it works 

data[data.TotalCharges > 1000] -> it shows me long list of error with last line 'ArrowInvalid: Failed to parse string: ' ' as a scalar of type float'

**Reply by JovanVeljanoski** [reply]
By reproducible example I mean a small code snippet that reproduces the issue. Something we can copy-paste in our local environment as see what happens. Like a unit-test basically. Thank would help. Feel free to submit a PR with a unit-test also if you are up for it.

**Reply by JovanVeljanoski** [reply]
Ah ok yeah something like that, but including a small aprt of the data. I can't run your example as it is.

**Reply by JovanVeljanoski** [reply]
Please just do a minimal reproducible example.

**Reply by JovanVeljanoski** [reply]
Ok you can track the progress of this in: 
https://github.com/vaexio/vaex/pull/1421

For future reference, this is the kind of code I was looking for:

```python
import vaex
df = vaex.from_dict({'x': ['1', '2.3', '', '', '5', '-5']})
df = df[~(df.x == '')]
df['x'] = df.x.astype('float')
df = df[df.x > 0]
```

If you provide things like this in the future, we can get to your problem faster, and be more motivated to fix things generally. Thanks.

**Reply by dharik-arsath** [reply]
import vaex
my_data = {'TotalCharges':[1000,2000,' ',' ',500]}
data = vaex.from_dict(my_data)
data = data[~(data.TotalCharges == ' ')]
data['TotalCharges'] = data.TotalCharges.astype('int16')
data[data.TotalCharges > 500]

# infact data.TotalCharges > 500 gives boolean indexing perfectly when I convert into dataframe it shows the error

I hope I can able to produce code as you want. Thanks
